sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -335,33 +333,62 @@ private[sbtplugin] object ScalaJSPluginInternal {
       scalaJSConfigSettings
   )
 
-  private val scalaJSTestFrameworkSettings = Seq(
+  val scalaJSTestSettings: Seq[Setting[_]] = (
+      scalaJSConfigSettings
+  ) ++ Seq(
+      /* Always default to false for scalaJSUseMainModuleInitializer in testing
+       * configurations, even if it is true in the Global configuration scope.
+       */
+      scalaJSUseMainModuleInitializer := false,
+
+      // Use test module initializer by default.
+      scalaJSUseTestModuleInitializer := true,
+
+      scalaJSModuleInitializers ++= {
+        if (scalaJSUseTestModuleInitializer.value) {
+          if (scalaJSUseMainModuleInitializer.value) {
+            val configName = configuration.value.name
+            throw new MessageOnlyException("You may only set one of " +
+                s"`scalaJSUseMainModuleInitializer in $configName` and " +
+                s"`scalaJSUseTestModuleInitializer in $configName` to true")
+          }
+
+          Seq(
+              ModuleInitializer.mainMethod(
+                  TestAdapterInitializer.ModuleClassName,
+                  TestAdapterInitializer.MainMethodName)
+          )
+        } else {
+          Seq.empty
+        }
+      },
+
       loadedTestFrameworks := {
+        val configName = configuration.value.name
+
         if (fork.value) {
           throw new MessageOnlyException(
-              "`test` tasks in a Scala.js project require " +
-              "`fork in Test := false`.")
+              s"`test in $configName` tasks in a Scala.js project require " +
+              s"`fork in $configName := false`.")
         }
 
-        val frameworks = testFrameworks.value
-        val env = jsEnv.value
-        val files = jsExecutionFiles.value
-
-        val moduleIdentifier = scalaJSLinkerConfig.value.moduleKind match {
-          case ModuleKind.NoModule =>
-            ModuleIdentifier.NoModule
-          case ModuleKind.CommonJSModule =>
-            ModuleIdentifier.CommonJSModule(scalaJSLinkedFile.value.data.getPath)
+        if (!scalaJSUseTestModuleInitializer.value) {
+          throw new MessageOnlyException(
+              s"You may only use `test in $configName` tasks in " +
+              "a Scala.js project, if `scalaJSUseTestModuleInitializer in " +
```

Comment:
Ah, I hadn't seen that before, but there shouldn't be a comma here ;)

(I haven't yet gotten used to all those commas when I read in German :sweat_smile:)

Reply:
Done.